addons/website_event_sale/models/sale_order.py

```diff
@@ -71,8 +71,10 @@ def _cart_update(self, product_id=None, line_id=None, add_qty=0, set_qty=0, **kw
             if ticket.id:
                 self = self.with_context(event_ticket_id=ticket.id, fixed_price=1)
         else:
-            line = None
-            ticket = self.env['event.event.ticket'].search([('product_id', '=', product_id)], limit=1)
+            if self.env.context.get("event_ticket_id"):
+                ticket = self.env['event.event.ticket'].search(['&', ('id', '=', self.env.context['event_ticket_id']), ('product_id', '=', product_id)], limit=1)
+            else:
+                ticket = self.env['event.event.ticket'].search([('product_id', '=', product_id)], limit=1)
```

Comment:
Sugg:

```py
ticket_domain = [('product_id', '=', product_id)]
if self.env.context.get("event_ticket_id"):
    ticket_domain = expression.AND([ticket_domain, [('id', '=', self.env.context['event_ticket_id'])]])

ticket = self.env['event.event.ticket'].search(ticket_domain, limit=1)
```

Also, does this issue not appear earlier than in v15?
Code seems similar in v14, did not dig earlier.

Reply:
@awa-odoo : After some test, this is indeed present in V13. I'll add another PR from V13 to v14 then and ping you on it

Reply:
Hi @anhe-odoo you can edit this PR so it targets 13.0, and rebase the code for 13.0 and push to the same branch (`15.0-opw-2761303-website_event_sale-fix_ticket_selection_in_sale_order-anhe`)